packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -226,6 +342,112 @@ function compileProofObject(objInfo: CompilerObjInfo<POD>): ObjectModuleInputs {
   };
 }
 
+function compileProofListMembership<
+  ObjInput extends POD | GPCRevealedObjectClaims
+>(
+  listConfig: Record<PODName, PODEntryIdentifier[]>,
+  listInput: Record<PODName, PODValue[] | PODValueTuple[]>,
+  entryMap: Map<PODEntryIdentifier, CompilerEntryInfo<ObjInput>>,
+  tupleMap: Map<TupleIdentifier, CompilerTupleInfo>,
+  paramTupleArity: number,
+  paramMaxLists: number,
+  paramMaxListElements: number
+): {
+  listComparisonValueIndex: CircuitSignal[];
+  listValidValues: CircuitSignal[][];
+} {
+  // Arrange list names alphabetically.
+  const listNameOrder = Object.keys(listConfig).sort();
+
+  // Do the same for the lists of comparison IDs and string them together.
```

Comment:
So given this list of pairs allows for duplicate list names, you actually are implicitly turning a single input list into multiple so as to support multiple entries/tuples?  I didn't expect that (as you can see in my comments elsewhere).  That's cool, though I think it needs more clarity in public docs (about what limits and flexibility is available to users) and in comments here (explaining what this code is doing).